controller/controller.go

```diff
@@ -864,11 +864,11 @@ func (ctrl *ProvisionController) Run(ctx context.Context) {
 	go ctrl.volumeStore.Run(ctx, DefaultThreadiness)
 
 	if ctrl.leaderElection {
-		rl, err := resourcelock.New("endpoints",
+		rl, err := resourcelock.New("leases",
 			ctrl.leaderElectionNamespace,
 			strings.Replace(ctrl.provisionerName, "/", "-", -1),
-			ctrl.client.CoreV1(),
 			nil,
+			ctrl.client.CoordinationV1(),
```

Comment:
@snOm3ad Could you please make me understand what was the reason for changing this argument.